src/dstack/_internal/server/services/fleets.py

```diff
@@ -137,6 +140,27 @@ async def get_plan(
 ) -> FleetPlan:
     # TODO: refactor offers logic into a separate module to avoid depending on runs
 
+    if spec.configuration.ssh_config and spec.configuration.ssh_config.hosts:
+        # there are manually listed hosts, need to check them for existence
+        active_instances = await list_active_remote_instances(session=session)
+
+        existing_hosts = set()
+        for instance in active_instances:
+            instance_conn_info = RemoteConnectionInfo.parse_raw(
+                cast(str, instance.remote_connection_info)
+            )
+            existing_hosts.add(instance_conn_info.host)
+
+        instances_already_in_fleet = []
+        for new_instance in spec.configuration.ssh_config.hosts:
```

Comment:
`ssh_config.hosts` may store hostnames but it may also store structs containing a hostname. You need to handle the second case.

https://github.com/dstackai/dstack/blob/19cd2fdc667f312720f5b228920870ecdc79892a/src/dstack/_internal/core/models/fleets.py#L64-L69